### PR TITLE
Refactored step defs for automated testing

### LINKED ISSUE #78: Step def functionality for testing
Discussed here https://github.com/EddieHubCommunity/api/pull/76#discussion_r619777611

I think we need to create another step def similar to this one that would make it easier to read - I can do it in another PR after this one is merged.

```ts
@then(/the response in item "([^"]*)" should contains:/)
public dataResponseItemTable(item: number, table: { rawTable: [] }) {
    const data = this.context.tableToObject(table);
    expect(JSON.parse(this.context.response.text)[item]).to.to.eql(data);
}
```
https://github.com/EddieHubCommunity/api/blob/main/test/step-definitions/responses.ts#L31

The new step def would be something like this

```ts
@then(/the response property "([^"]*)" in field "([^"]*)" should contains:/)
public dataResponseItemTable(property: string, item: number, table: { rawTable: [] }) {
    const data = this.context.tableToObject(table);
    expect(JSON.parse(this.context.response.text)[property][item]).to.eql(data);
}
```

The tests would be something like

```gherkin
And  the response property "future" in item "0" should contains:
            | id   | 123                |
            | name | "Livestream XY"    |
And  the response property "ongoing" in item "0" should contains:
            | id   | 123                |
            | name | "Livestream YZ"    |
```

Note: I think `contains` should be singular 🤔 - this can be changed later

### PR DESCRIPTION
**Pull request by eddiejaoude**
closes #78

Example

```diff
- And the response should contains:
-             | statusCode | 400                                                                                                                                                                                                                                                                                                                                              - |
-             | message    | ["name must be a string","name should not be empty","platform must be one of the following values: - YouTube, Twitch","platform must be a string","platform should not be empty","author should not be empty","startDate should - not be empty","startDate must be a Date instance","endDate must be a Date instance","endDate should not be empty"] |
-             | error      | "Bad Request"   

+ And the response should contain:
+             | statusCode | 400           |
+             | error      | "Bad Request" |
+         And the response property "message" has items:
+             | name must be a string                                         |
+             | name should not be empty                                      |
+             | platform must be one of the following values: YouTube, Twitch |
+             | platform must be a string                                     |
+             | platform should not be empty                                  |
+             | author should not be empty                                    |
+             | startDate should not be empty                                 |
+             | startDate must be a Date instance                             |
+             | endDate must be a Date instance                               |
+             | endDate should not be empty                                   |
```

### CODE REVIEW

File: test/features/calendar.feature [hunk 1]
```diff
@@ -36,10 +36,20 @@ Feature: calendar module
...
-            | message    | ["name must be a string","name should not be empty","platform must be one of the following values: YouTube, Twitch","platform must be a string","platform should not be empty","author should not be empty","startDate should not be empty","startDate must be a Date instance","endDate must be a Date instance","endDate should not be empty"] |
-            | error      | "Bad Request"                                                                                                                                                                                                                                                                                                                                    |
+        And the response should contain:
+            | statusCode | 400           |
+            | error      | "Bad Request" |
+        And the response property "message" has items:
+            | name must be a string                                         |
```
Comment (Cahllagerfeld) [inline test/features/calendar.feature:43]: We should document somewhere that we dont use quotes here.

**Comment by Cahllagerfeld** [comment]
I really like that it gets more readable now! 😄

File: test/features/calendar.feature (hunk 1 above)
Comment (eddiejaoude) [reply test/features/calendar.feature:43]: Good point 👍 

We should probably document the whole testing process with some simple examples - I will raise an issue

Comment (eddiejaoude) [reply test/features/calendar.feature:43]: Issue created https://github.com/EddieHubCommunity/api/issues/82

**Review by Cahllagerfeld — APPROVED** [review]
I think these changes are fine then! 💥

**Comment by eddiejaoude** [comment]
Awesome, thank you 👍

**Pull request merged**